be/test/vec/aggregate_functions/agg_group_array_test.cpp

```diff
@@ -0,0 +1,142 @@
+        using FieldType = typename DataType::FieldType;
+        auto type = std::make_shared<DataType>();
+        auto input_col = type->create_column();
+        for (size_t i = 0; i < input_nums; ++i) {
+            if constexpr (std::is_same_v<DataType, DataTypeString>) {
+                auto item = std::string("item") + std::to_string(i);
+                input_col->insert_data(item.c_str(), item.size());
+            } else {
+                auto item = FieldType(static_cast<uint64_t>(i));
+                input_col->insert_data(reinterpret_cast<const char*>(&item), 0);
+            }
+        }
+        EXPECT_EQ(input_col->size(), input_nums);
+
+        const IColumn* column[1] = {input_col.get()};
+        for (int i = 0; i < input_col->size(); i++) {
+            agg_function->add(place, column, i, &_agg_arena_pool);
+        }
+    }
+
+    template <typename DataType>
+    void test_agg_group_uniq_array(size_t input_nums = 0) {
+        DataTypes data_types = {(DataTypePtr)std::make_shared<DataType>()};
+        LOG(INFO) << "test_agg_group_uniq_array for type"
+                  << "(" << data_types[0]->get_name() << ")";
+
+        Array array;
+        AggregateFunctionSimpleFactory factory = AggregateFunctionSimpleFactory::instance();
+        auto agg_function = factory.get("group_uniq_array", data_types, array);
+        EXPECT_NE(agg_function, nullptr);
+
+        std::unique_ptr<char[]> memory(new char[agg_function->size_of_data()]);
+        AggregateDataPtr place = memory.get();
+        agg_function->create(place);
+
+        agg_group_uniq_array_add_elements<DataType>(agg_function, place, input_nums);
+
+        ColumnString buf;
+        VectorBufferWriter buf_writer(buf);
+        agg_function->serialize(place, buf_writer);
+        buf_writer.commit();
+        VectorBufferReader buf_reader(buf.get_data_at(0));
+        agg_function->deserialize(place, buf_reader, &_agg_arena_pool);
+
+        std::unique_ptr<char[]> memory2(new char[agg_function->size_of_data()]);
+        AggregateDataPtr place2 = memory2.get();
+        agg_function->create(place2);
+
+        agg_group_uniq_array_add_elements<DataType>(agg_function, place2, input_nums);
+
+        agg_function->merge(place, place2, &_agg_arena_pool);
+        auto column_result = ColumnArray::create(data_types[0]->create_column());
+        agg_function->insert_result_into(place, *column_result);
+        EXPECT_EQ(column_result->size(), 1);
+        EXPECT_EQ(column_result->get_offsets()[0], input_nums);
+
+        auto column_result2 = ColumnArray::create(data_types[0]->create_column());
+        agg_function->insert_result_into(place2, *column_result2);
+        EXPECT_EQ(column_result2->size(), 1);
+        EXPECT_EQ(column_result->get_offsets()[0], input_nums);
+
+        LOG(INFO) << column_result->get_offsets()[0];
+        LOG(INFO) << column_result2->get_offsets()[0];
```

Comment:
are these two `LOG` statements useless?

Reply:
Added for dubugging ,I will remove them.